File: Templates/Java/integrationtesting/src/main/java/com/example/integrationtesting/BaseUser.java
package com.example.integrationtesting;

import java.util.Optional;

public interface BaseUser {
        int id();
        String name();
        String username();
        String email();
        String phone();
        String website();
        Company company();
}

record User(int id, String name, String username, String email, String phone, String website, Company company) implements BaseUser {}

record AddressUser(int id, String name, String username, String email, Address address, String phone, String website, Company company) implements BaseUser {}

record Address(String street, String suite, String city, String zipcode, Optional<Geo> geo) {}

record Company(String name, String catchPhrase, String bs) {}

record Geo(double lat, double lng) {}

File: Templates/Kotlin/integrationtesting/src/main/java/com/example/integrationtesting/BaseUser.kt
package com.example.integrationtesting

import java.util.*

@JvmRecord
data class User(
        @JvmField val id: Int,
        val name: String?,
        val username: String?,
        val email: String?,
        val phone: String?,
        val website: String?,
        val company: Company?
)

@JvmRecord
data class AddressUser(
        @JvmField val id: Int,
        val name: String?,
        val username: String?,
        val email: String?,
        @JvmField val address: Address?,
        val phone: String?,
        val website: String?,
        val company: Company?
)

@JvmRecord
data class Address(
        val street: String?,
        val suite: String?,
        val city: String?,
        val zipcode: String?,
        @JvmField val geo: Optional<Geo?>?
)

@JvmRecord
data class Company(val name: String?, val catchPhrase: String?, val bs: String?)

@JvmRecord
data class Geo(@JvmField val lat: String, @JvmField val lng: String)